lib/r10k/action/runner.rb

```diff
@@ -100,11 +100,26 @@ def settings_from_config(override_path)
       def add_credential_overrides(overrides)
         sshkey_path = @opts[:'private-key']
         token_path = @opts[:'oauth-token']
+        app_id = @opts[:'github-app-id']
+        app_private_key_path = @opts[:'github-app-key']
+        app_ttl = @opts[:'github-app-ttl']
 
         if sshkey_path && token_path
           raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
         end
 
+        if sshkey_path && app_private_key_path
+          raise R10K::Error, "Cannot specify both a SSH key and a SSL key to use with this deploy."
+        end
+
+        if app_id && token_path
+          raise R10K::Error, "Cannot specify both a Github App and a token to use with this deploy."
+        end
```

Comment:
Would it make sense to check that `app_id` OR `app_private_key_path` is not specified alongside `sshkey_path` and `token_path`?